controller/internal/enforcer/nfqdatapath/datapath_tcp.go

```diff
@@ -1120,7 +1120,7 @@ func (d *Datapath) netSynRetrieveState(p *packet.Packet) (*connection.TCPConnect
 
 	context, err := d.contextFromIP(false, p.Mark, p.DestPort(), packet.IPProtocolTCP)
 	if err == nil {
-		if conn, err := d.netOrigConnectionTracker.GetReset(p.L4FlowHash(), 0); err == nil && !conn.(*connection.TCPConnection).MarkForDeletion {
+		if conn, err := d.netOrigConnectionTracker.GetReset(p.L4FlowHash(), 0); err == nil && !conn.(*connection.TCPConnection).MarkForDeletion && conn.(*connection.TCPConnection).GetInitialSequenceNumber() == p.TCPSequenceNumber() {
```

Comment:
why does MarkForDeletion not need a lock to access but isn does inside. a function called GetISN ?

Reply:
removed the lock from sequence number i had provided a set so added a lock. Removed the set function as well 
We are setting this only on create so no lock required. 
The markforDeleteion is modified under lock ideally this place should have a Rlock on it

Reply:
Added Rlock for MarkForDeletion flag. the set happens under a global flag which is locked